pkg/karmadactl/util/check.go

```diff
@@ -0,0 +1,84 @@
+package util
+
+import (
+	"context"
+	"time"
+
+	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+)
+
+// WaitForStatefulsetRollout wait statefulset rollout
+func WaitForStatefulsetRollout(c kubernetes.Interface, sts *appsv1.StatefulSet, timeout int) error {
+	var lastErr error
+	if err := wait.Poll(time.Second, time.Duration(timeout)*time.Second, func() (bool, error) {
+		s, err := c.AppsV1().StatefulSets(sts.GetNamespace()).Get(context.TODO(), sts.GetName(), metav1.GetOptions{})
+		if err != nil {
+			lastErr = err
+			klog.Warningf("WaitForStatefulsetRollout: failed to get StatefulSet %s/%s", sts.GetNamespace(), sts.GetName())
+			return false, nil
+		}
+		if s.Generation > s.Status.ObservedGeneration {
+			lastErr = errors.Errorf("expected generation %d, observed generation: %d",
```

Comment:
Could we directly use `fmt.Errorf`?

`errors.Errorf` seems out of day, refer to https://stackoverflow.com/questions/61933650/whats-the-difference-between-errors-wrapf-errors-errorf-and-fmt-errorf

Reply:
good point. `errors` package has been deprecated.

Reply:
good point. i fixed it